platform/security/encryption/platform-security-encryption-crypter/src/main/java/ddf/security/encryption/crypter/Crypter.java

```diff
@@ -0,0 +1,277 @@
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package ddf.security.encryption.crypter;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.crypto.tink.Aead;
+import com.google.crypto.tink.CleartextKeysetHandle;
+import com.google.crypto.tink.JsonKeysetReader;
+import com.google.crypto.tink.JsonKeysetWriter;
+import com.google.crypto.tink.KeysetHandle;
+import com.google.crypto.tink.aead.AeadConfig;
+import com.google.crypto.tink.aead.AeadFactory;
+import com.google.crypto.tink.aead.AeadKeyTemplates;
+import ddf.security.SecurityConstants;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.security.AccessController;
+import java.security.GeneralSecurityException;
+import java.security.PrivilegedAction;
+import java.security.SecureRandom;
+import java.util.Base64;
+import java.util.Properties;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * A "crypter" combines an "encryptor" and a "decryptor". This class is responsible for abstracting
+ * underlying cryptography implementations for use throughout the system.
+ */
+public class Crypter {
+  private static final Logger LOGGER = LoggerFactory.getLogger(Crypter.class);
+  private static final String DEFAULT_KEYSET_FILE_NAME = "default";
+  private static final String KEYSET_FILE_EXTENSION = ".json";
+  private static final int ASSOCIATED_DATA_BYTE_SIZE = 10;
+
+  private final String keysetDir =
+      AccessController.doPrivileged(
+          (PrivilegedAction<String>) () -> System.getProperty(SecurityConstants.KEYSET_DIR));
+  private final String associatedDataPath =
+      AccessController.doPrivileged(
+          (PrivilegedAction<String>)
+              () -> System.getProperty(SecurityConstants.ASSOCIATED_DATA_PATH));
+
+  @VisibleForTesting KeysetHandle keysetHandle;
+  private byte[] associatedData;
+  private Aead aead;
+
+  /**
+   * Read/create a specified keyset under the keyset directory.
+   *
+   * @param keysetFileName name of the keyset to read/create.
+   * @throws CrypterException
+   */
+  public Crypter(String keysetFileName) throws CrypterException {
+    String keysetLocation = Paths.get(keysetDir, keysetFileName + KEYSET_FILE_EXTENSION).toString();
+
+    synchronized (Crypter.class) {
```

Comment:
✏️ I recall there was a reason to synchronize access here before, but there's definitely not shared data that can be read/written by another thread here now. Your static fields are all final and no other thread could be touching the `keysetHandle`, `associatedData`, or `aead` fields until after this ctor has completed processing.

Reply:
Here is my reasoning for keeping the synchronized block. Let me know what you think 😄.

The constructor undergoes logic that will read from/write to a keyset file and an associated data file. If two different threads are calling this constructor at the same time, there could be concurrency issues regarding those files.